### PR TITLE
fix py3 bytes/str issue when viewing a MARC record

### PR DESCRIPTION
**Pull request by hornc**
Fixes the following errors:
```
<class 'TypeError'> at /show-records/marc_loc_2016/BooksAll.2016.part41.utf8:32007375:748
argument should be integer or bytes-like object, not 'str'
```
URL: 

https://openlibrary.org/show-records/marc_loc_2016/BooksAll.2016.part41.utf8:32007375:748
https://openlibrary.org/show-records/marc_loc_2016/BooksAll.2016.part41.utf8:32007375:748?debug=true

Closes #

### CODE REVIEW

**Review by cclauss — APPROVED** [review]
LGTM.  Changes made to deprecated functions.

**Review by cdrini — CHANGES_REQUESTED** [review]
Trusting you know what you're doing by removing the code/error checks that you did; no clue why they're necessary :P 

Tested:
- [x] https://dev.openlibrary.org/show-records/marc_loc_2016/BooksAll.2016.part41.utf8:32007375:748
- [ ] This is still erroring: http://staging.openlibrary.org/show-records/marc_loc_2016/BooksAll.2016.part41.utf8:32007375:748?debug=true

**Comment by cclauss** [comment]
Please do integer division on line 175 (like #4037)
```python
iter_dir = (directory[i*12:(i+1)*12] for i in range(len(directory) / 12))
# —>
iter_dir = (directory[i*12:(i+1)*12] for i in range(len(directory) // 12))
```

**Review by mekarpeles — APPROVED** [review]

**Comment by mekarpeles** [comment]
The changes look fine but as noted, the error still exists

```
<class 'TypeError'> at /show-records/marc_loc_2016/BooksAll.2016.part41.utf8:32007375:748
'float' object cannot be interpreted as an integer

Python | /openlibrary/openlibrary/catalog/marc/fast_parse.py in read_directory, line 175
GET http://staging.openlibrary.org/show-records/marc_loc_2016/BooksAll.2016.part41.utf8:32007375:748
```

**Comment by hornc** [comment]
thanks @cdrini for finding that failing case, good catch! This should fix it.

File: openlibrary/catalog/marc/fast_parse.py [hunk 1]
```diff
@@ -173,27 +173,22 @@ def read_directory(data):
...
-        last = offset+length
-        if data[last] != '\x1e':
-            length += data[last:].find('\x1e')
-    except IndexError:
-        pass
+    if data[offset] != b'\x1e':
+        offset += data[offset:].find(b'\x1e')
+    last = offset + length
+    if data[last] != b'\x1e':
+        length += data[last:].find('b\x1e')
```
Comment (cdrini) [inline openlibrary/catalog/marc/fast_parse.py:190]: :P
```suggestion
        length += data[last:].find(b'\x1e')
```

**Comment by cdrini** [comment]
Ahh, still getting another error here :P 

http://staging.openlibrary.org/show-records/marc_loc_2016/BooksAll.2016.part41.utf8:32007375:748?debug=true

**Comment by hornc** [comment]
@cdrini That should be fixed now. I'm unable to test locally because my docker env won't build with the recent python 3.9 changes.
I was getting an error about not being able to find python 3.9 when building the covers image, which seemed strange since it was there for the `olbase` image.

```
Step 9/13 : RUN python3.9 -m pip install -r requirements_test.txt
 ---> Running in 0a034c8d7ae4
/bin/sh: 1: python3.9: not found
ERROR: Service 'covers' failed to build : The command '/bin/sh -c python3.9 -m pip install -r requirements_test.txt' returned a non-zero code: 127
```

**Comment by cdrini** [comment]
Oh! :( You might want to try doing a "full" re-build: `docker-compose build --pull`; that will also pull the latest `openlibrary/olbase` image from docker hub.

**Comment by cdrini** [comment]
I'll toss your changes on to staging now as well to see if this fixes the issue!

**Comment by cdrini** [comment]
Turtles all the way down 😅 http://staging.openlibrary.org/show-records/marc_loc_2016/BooksAll.2016.part41.utf8:32007375:748?debug=true

**Comment by cdrini** [comment]
Refreshed on staging; there's more 😅 http://staging.openlibrary.org/show-records/marc_loc_2016/BooksAll.2016.part41.utf8:32007375:748?debug=true

Any luck with the docker commands?

**Comment by cclauss** [comment]
Use Py 3.8.6, not 3.9.0.

**Comment by hornc** [comment]
@cdrini `docker-compose build --pull` did the trick, thank you.

I can run everything locally and view that last link:
image

When I tried it on staging earlier today I _did_ see an error:
```
Python | /openlibrary/openlibrary/plugins/upstream/code.py in static_url, line 89
-- | --
GET http://staging.openlibrary.org/show-records/marc_loc_2016/BooksAll.2016.part41.utf8:32007375:748
```

But I'm not getting in now, not on staging nor locally. I'm runnin geverything with
`PYENV_VERSION=3.8.6 docker-compose up`

**Comment by hornc** [comment]
Last commit fixes the `b'001'` display of the MARC fields:

image

**Comment by cdrini** [comment]
Woohoo! Working on staging: staging.openlibrary.org/show-records/marc_loc_2016/BooksAll.2016.part41.utf8:32007375:748 , too :)

**Comment by hornc** [comment]
@cdrini ok, tests finally match desired behaviour, they went through a stage of passing with the incorrect `b'100'` output... <sigh>
I

**Review by cdrini — APPROVED** [review]
Awesome! Lgtm

**Pull request merged**